include/seqan3/utility/bloom_filter/bloom_filter.hpp

```diff
@@ -0,0 +1,397 @@
+            data[idx] = 1;
+        };
+    }
+
+    /*!\brief Remove all values from the Bloom Filter by setting all bits to 0.
+     *
+     * \attention This function is only available for **uncompressed** Bloom Filters.
+     *
+     * \details
+     *
+     * While all values are removed from the vector, the size of the Bloom Filter is not changed.
+     * 
+     * ### Example
+     *
+     * \include test/snippet/utility/bloom_filter/bloom_filter_reset.cpp
+     */
+    void reset() noexcept
+    //!\cond
+        requires (data_layout_mode == data_layout::uncompressed)
+    //!\endcond
+    {
+        sdsl::util::_set_zero_bits(data);
+    }
+    //!\}
+    
+    /*!\name Lookup
+     * \{
+     */
+    /*!\brief Check whether a value is present in the Bloom Filter.
+     * \param[in] value The raw numeric value to process.
+     *
+     * \attention This function is only available for **uncompressed** Bloom Filters.
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/utility/bloom_filter/bloom_filter_contains.cpp
+     */
+    bool contains(size_t const value) const noexcept
+    {
+        for ( size_t i = 0; i < hash_funs; i++ )
+        {
+            size_t idx = hash_and_fit(value, hash_seeds[i]);
+            assert(idx < data.size());
+            if ( data[idx] == 0 )
+                return false;
+        }
+        return true;
+    }
+    //!\}
+
+    /*!\name Counting
+     * \{
+     */
+    /*!\brief Counts the occurrences for all values in a range.
+     * \tparam value_range_t The type of the range of values. Must model std::ranges::input_range. The reference type
+     *                       must model std::unsigned_integral.
+     * \param[in] values The range of values to process.
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/utility/bloom_filter/bloom_filter_count.cpp
+     *
+     * ### Thread safety
+     *
+     * Concurrent invocations of this function are thread safe.
+     */
+    template <std::ranges::range value_range_t>
+    size_t count(value_range_t && values) const noexcept
+    {
+        static_assert(std::ranges::input_range<value_range_t>, "The values must model input_range.");
+        static_assert(std::unsigned_integral<std::ranges::range_value_t<value_range_t>>,
+                      "An individual value must be an unsigned integral.");
+
+        size_t result = 0;
+
+        for (auto && value : values)
+            result += contains(value);
+
+        return result;
+    }
+    //!\}
+
+    /*!\name Capacity
+     * \{
+     */
+    /*!\brief Returns the number of hash functions used in the Bloom Filter.
+     * \returns The number of hash functions.
+     */
+    size_t hash_function_count() const noexcept
+    {
+        return hash_funs;
+    }
+
+    /*!\brief Returns the size that the Bloom Filter manages.
+     * \returns The size in bits.
+     */
+    size_t size() const noexcept
```

Comment:
```suggestion
    size_t bit_size() const noexcept
```

`size` is kind of a reserved keyword for the number of elements.

Can we use the same description as in:

https://github.com/seqan/seqan3/blob/c7d7f1a60d8946a09cc8ca3405550a88c79d8b12/include/seqan3/search/dream_index/interleaved_bloom_filter.hpp#L479-L485

?